cli/doc/mod.rs

```diff
@@ -0,0 +1,64 @@
+pub mod class;
+pub mod r#enum;
+pub mod function;
+pub mod interface;
+pub mod module;
+pub mod namespace;
+mod node;
+pub mod parser;
+mod printer;
+pub mod ts_type;
+pub mod type_alias;
+pub mod variable;
+
+pub use node::DocNode;
+pub use node::DocNodeKind;
+pub use node::Location;
+pub use node::ParamDef;
+pub use parser::DocParser;
+pub use printer::JSONPrinter;
+pub use printer::TerminalPrinter;
+
+#[cfg(test)]
+mod tests;
+
+pub fn find_node_by_name_recursively(
+  doc_nodes: Vec<DocNode>,
+  name: String,
+) -> Option<DocNode> {
+  let mut parts = name.splitn(2, '.');
+  let name = parts.next();
+  let leftover = parts.next();
+  name?;
+  let node = find_node_by_name(doc_nodes, name.unwrap().to_string());
+  match node {
+    Some(node) => match node.kind {
+      DocNodeKind::Namespace => {
+        if let Some(leftover) = leftover {
+          find_node_by_name_recursively(
+            node.namespace_def.unwrap().elements,
+            leftover.to_string(),
+          )
+        } else {
+          Some(node)
+        }
+      }
+      _ => {
+        if leftover.is_none() {
+          Some(node)
+        } else {
+          None
+        }
+      }
+    },
+    _ => None,
+  }
+}
+
+fn find_node_by_name(doc_nodes: Vec<DocNode>, name: String) -> Option<DocNode> {
+  let node = doc_nodes.iter().find(|node| node.name == name);
```

Comment:
Maybe this would be more useful if the matching was similar to how `cargo test $FILTER` works. That is, instead use `node.name.contains(name)` ?

Reply:
It's a bit tricky because of namespaces... Can we leave it as is for first iteration and change in follow ups?